src/home/rendering/render-github-notifications.ts

```diff
@@ -0,0 +1,301 @@
+import { organizationImageCache } from "../fetch-github/fetch-data";
+import { GitHubAggregated } from "../github-types";
+import { renderErrorInModal } from "./display-popup-modal";
+import { getTimeAgo } from "./utils";
+import { notificationsContainer } from "../home";
+
+export async function renderNotifications(notifications: GitHubAggregated[], skipAnimation: boolean) {
+  if (notificationsContainer.classList.contains("ready")) {
+    notificationsContainer.classList.remove("ready");
+    notificationsContainer.innerHTML = "";
+  }
+  const existingNotificationIds = new Set(
+    Array.from(notificationsContainer.querySelectorAll(".issue-element-inner")).map((element) => element.getAttribute("data-issue-id"))
+  );
+
+  let delay = 0;
+  const baseDelay = 1000 / 15; // Base delay in milliseconds
+
+  for (const notification of notifications) {
+    if (!existingNotificationIds.has(notification.notification.id.toString())) {
+      const issueWrapper = await everyNewNotification({ notification: notification, notificationsContainer });
+      if (issueWrapper) {
+        if (skipAnimation) {
+          issueWrapper.classList.add("active");
+        } else {
+          setTimeout(() => issueWrapper.classList.add("active"), delay);
+          delay += baseDelay;
+        }
+      }
+    }
+  }
+  notificationsContainer.classList.add("ready");
+  // Call this function after the issues have been rendered
+  //setupKeyboardNavigation(notificationsContainer);
+
+  // Scroll to the top of the page
+  window.scrollTo({ top: 0 });
+}
+export function renderEmpty() {
+  if (notificationsContainer.classList.contains("ready")) {
+    notificationsContainer.classList.remove("ready");
+    notificationsContainer.innerHTML = "";
+  }
+  const issueWrapper = document.createElement("div");
+  issueWrapper.style.marginTop = "20px";
+  const issueElement = document.createElement("div");
+  issueElement.innerHTML = `
+    <div class="info"><div class="title"><h3>No notifications found</h3></div></div>
+  `;
+  issueElement.classList.add("issue-element-inner");
+  issueWrapper.appendChild(issueElement);
+  notificationsContainer.appendChild(issueWrapper);
+
+  issueWrapper.classList.add("active");
+  notificationsContainer.classList.add("ready");
+}
+
+async function everyNewNotification({ notification, notificationsContainer }: { notification: GitHubAggregated; notificationsContainer: HTMLDivElement }) {
+  const issueWrapper = document.createElement("div");
+  const issueElement = document.createElement("div");
```

Comment:
If this is looping, then you should be cloning instead of creating new elements every time it's a lot more efficient that way